File: apps/api/aqapi/shared/entity/currency_by_language_entity.py
from datetime import datetime
from typing import override
from sqlalchemy import Column, ForeignKey, Integer, String, Boolean, DateTime, func
from sqlalchemy.orm import relationship, Mapped, mapped_column
from aqapi.core.entity.base_entity import BaseEntity
from aqapi.core.entity.base_translation_entity import BaseTranslationEntity
from aqapi.core.config.db_config import DB_CONF


class CurrencyByLanguageEntity(BaseEntity):
    """通貨と言語の関連エンティティ"""

    __tablename__ = "currencies_by_language"

    currency_id: Mapped[int] = mapped_column(Integer, ForeignKey("currencies.id", ondelete="CASCADE"), nullable=False, comment="通貨コード(外部キー：currencies.id")
    language_id: Mapped[int] = mapped_column(Integer, ForeignKey("languages.id", ondelete="CASCADE"), nullable=False, comment="言語コード")

    # Relationships
    currency = relationship("CurrenciesEntity", foreign_keys=[currency_id])
    language = relationship("LanguagesEntity", foreign_keys=[language_id])

    @override
    @classmethod
    def _seed_data(cls) -> list[BaseEntity]:
        return [
            CurrencyByLanguageEntity(currency_id=1, language_id=1),  # JPY
            CurrencyByLanguageEntity(currency_id=2, language_id=2),  # USD
        ]

File: packages/backend/src/features/shared/entity/currencyByLanguageEntity.ts
import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  ManyToOne,
  JoinColumn,
  Unique,
  DataSource,
} from "typeorm";
import { CurrencyEntity } from "./currencyEntity";
import { LanguageEntity } from "../../language/entity/languageEntity";
import { BaseMasterEntity } from "@backend/core/entity/baseMasterEntity";

@Entity("currencies_by_language")
@Unique("uq_currency_by_language", ["currencyId", "languageId"])
export class CurrencyByLanguageEntity extends BaseMasterEntity {
  @PrimaryGeneratedColumn({ type: "int", comment: "ID" })
  id!: number;
  @Column({ name: "currency_id", type: "int", nullable: false, comment: "通貨ID(外部キー：currencies.id)" })
  currencyId!: number;
  @Column({ name: "language_id", type: "int", nullable: false, comment: "言語ID" })
  languageId!: number;

  // Relations
  @ManyToOne(() => CurrencyEntity)
  @JoinColumn({ name: "currency_id" })
  currency?: CurrencyEntity;
  @ManyToOne(() => LanguageEntity)
  @JoinColumn({ name: "language_id" })
  language?: LanguageEntity;

  /**
   * シードデータ
   */
  protected static seedData(): CurrencyByLanguageEntity[] {
    return [
      Object.assign(new CurrencyByLanguageEntity(), { currencyId: 1, languageId: 1 }), // JPY - Japanese
      Object.assign(new CurrencyByLanguageEntity(), { currencyId: 2, languageId: 2 }), // USD - English
    ];
  }
}
